Add link to plugin developers guide from README (#1417)
jenkinsci/configuration-as-code-plugin: README.md

Most plugins should be supported out-of-the-box, or maybe require some minimal changes. See this [dashboard](https://issues.jenkins.io/secure/Dashboard.jspa?selectPageId=18341) for known compatibility issues.

> ## Plugin developers

Author: Should I change the title to "Other plugins' developers"? Or "Third party plugin developers", or whatever you think it's best.

Reviewer: what about?
```suggestion
## Adding JCasC support to a plugin
```

> Plugin developers willing to support JCasC in their plugin should [check out our guidelines](docs/PLUGINS.md).

Reviewer: I think this should be:

```suggestion
Plugin developers willing to support JCasC in their plugin should [check out our guidelines](docs/CONTRIBUTING.md).
```

Reviewer: ah i see what you mean

Reviewer: ah i see what you mean

Reviewer: ```suggestion
Plugin developers wanting to support JCasC in their plugin should [check out our how-to guide](docs/PLUGINS.md).
```
